### PR TITLE
feat: upgrade for controller and e2e script

### PR DESCRIPTION
**Pull request by dreamlandliu**
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/domain/upgrade/step/UpdateK8sImage.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    public UpdateK8sImage(UpgradeAccess upgradeAccess, K8sClient k8sClient) {
+        super(upgradeAccess);
+        this.k8sClient = k8sClient;
+    }
+
+    @Override
+    protected void doStep(Upgrade upgrade) {
+        log.info("Update image" + upgrade.getTo());
+        JSONArray patchJson = new JSONArray(List.of(
```
Comment (goldenxinxing) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/upgrade/step/UpdateK8sImage.java:50]: use `String patchJson = JSONUtil.toJson(List.of(
                Map.of(
                        "op", "replace",
                        "path", "/spec/template/spec/containers/0/image",
                        "value", Objects.requireNonNull(upgrade.getTo().getImage())
                )
        ));` ?

File: scripts/e2e_test/upgrade_test.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+fi
+
+SW_USER="${SW_USER:=starwhale}"
+SW_PWD="${SW_PWD:=abcd1234}"
+HOST_URL="${HOST_URL:=upgrade-test.pre.intra.starwhale.ai}"
+IMAGE="$NEXUS_HOSTNAME:$PORT_NEXUS_DOCKER/star-whale/server:$SERVER_RELEASE_VERSION"
+
+HEADER_AUTH=$(curl -s -D - http://$HOST_URL/api/v1/login -d 'userName='${SW_USER}'&userPwd='${SW_PWD} | grep Authorization:)
+
+RESULT=$(curl -s -D - http://$HOST_URL/api/v1/system/version/upgrade -H "Content-Type: application/json" -H "${HEADER_AUTH}" -d '{"version": "'"${SERVER_RELEASE_VERSION}"'", "image": "'"${IMAGE}"'"}')
```
Comment (goldenxinxing) [inline scripts/e2e_test/upgrade_test.sh:15]: Whether there need a while loop until result is true?

File: server/controller/src/main/java/ai/starwhale/mlops/domain/upgrade/step/UpdateK8sImage.java (hunk 1 above)
Comment (dreamlandliu) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/upgrade/step/UpdateK8sImage.java]: done

File: scripts/e2e_test/upgrade_test.sh (hunk 2 above)
Comment (dreamlandliu) [reply scripts/e2e_test/upgrade_test.sh]: done

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**